pkg/ingester/shipper.go

```diff
@@ -252,7 +252,12 @@ func (s *Shipper) blockMetasFromOldest() (metas []*metadata.Meta, _ error) {
 func readShippedBlocks(dir string) (map[ulid.ULID]time.Time, error) {
 	meta, err := readShipperMetaFile(dir)
 	if err != nil {
```

Comment:
nit but, this whole block could be simplified a bit I think:
```go
if err != nil && errors.Is(err, os.ErrNotExist) {
    meta = shipperMeta{}
} else if err != nil {
    return nil, err
}
```

Reply:
I am taking @krajorama suggestion which looks more cleaner.

```
if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			// If the meta file doesn't exist it means the shipper hasn't run yet.
			meta = shipperMeta{}
		} else  {
			return nil, err
		}
	}
```